nengo/spa/vocab.py

```diff
@@ -147,6 +149,17 @@ def add(self, key, p):
 
         The pointer value can be a SemanticPointer or a vector.
         """
+        if self.read_only:
+            if self.parent is None:
+                # If this vocabulary is not a subset vocabulary, throw an error
+                raise ValueError("Cannot add semantic pointer '%s' to "
+                                 "read-only vocabulary." % key)
+            else:
+                # If this vocabulary is a subset vocabulary, apply operation on
+                # parent vocab.
+                self.parent.add(key, p)
+                return
```

Comment:
As per the discussion in the meeting, I think the solution for now is to adjust this PR such that if it's read only then it always raises the ValueError, but then also modify transform_to so that it handles that gracefully (basically if a vocab is read only and it doesn't have a key, then skip that key rather than trying to generate it).